fame/core/module.py

```diff
@@ -1108,3 +1113,59 @@ def static_info(cls):
         init_config_values(info)
 
         return ModuleInfo(info)
+
+
+class PreloadingModule(Module):
+    """ PreloadingModules can be used to download the sample
+        binary from e.g. VirusTotal before queueing any
+        processing modules. Hence, PreloadingModules only work
+        on hashes. A successful execution of a PreloadingModule
+        updates the Analysis object with the new data and queues
+        the remaining modules as if the sample itself was uploaded
+        the FAME.
+    """
+
+    queue = 'unix'
+
+    def __init__(self, with_config=True):
+        Module.__init__(self, with_config)
+        self.results = None
+        self.tags = []
+
+    def preload(self, target):
+        """ To implement.
+
+        Args:
+            target (string): the hash that is to be analyzed
+        Returns:
+            A boolean indicating whether or not a file could be downloaded
+            for the given hash
+        """
+        raise NotImplementedError
+
+    def add_preloaded_file(self, filepath, fd):
+        self._analysis.add_preloaded_file(filepath, fd)
+
+    def init_options(self, options):
+        for option in options:
+            setattr(self, option, options[option])
+
+    def execute(self, analysis):
+        self._analysis = analysis
+        self.init_options(analysis['options'])
+        return self.preload(self._analysis.get_main_file())
```

Comment:
There should be a try catch here so that any exception happening in a preloading module does not prevent the other preloading modules to execute.

Reply:
I added some try/catch logic